pkg/innerring/innerring.go

```diff
@@ -573,8 +575,11 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper) (*Server, error
 			settlementProcessor.HandleAuditEvent,
 		),
 		AlphabetSyncHandler: alphaSync,
-		NodeValidator:       locodeValidator,
-		FeeProvider:         server.feeConfig,
+		NodeValidator: nodevalidator.New(
+			locodeValidator,
+			addrvalidator.New(),
```

Comment:
As a micro optimization I suggest to place lighter weight validators at the beginning. Unfortunately, you have to evaluate by eye,

Reply:
agree, moved